src/main/resources/cdm/v5/optional_diff_V5_0.json

```diff
@@ -0,0 +1,2 @@
+{
```

Comment:
I don't like the fact, that we have a separate file for optional diff. I would suggest changing the existing diff. I see two options:
* add an section with option fields:
```
  "optionals_diff" : {
    "cost" : [ "reveue_code_source_value" ],
    "measurement" : [ "measurement_time" ]
  }
```
* or use ~ for such fields.

Reply:
I would peak the second options, and modify `calcDifference` method this way: 
```
    private Diff calcDifference(String expectedSchemaResource, Map<String, List<String>> databaseSchema) {

        Map<String, List<String>> expectedSchema = parseSchemaJson(expectedSchemaResource);

        Map<String, List<String>> tableColumns = new HashMap<>();
        Map<String, List<String>> optionalTableColumns = new HashMap<>();

        expectedSchema.forEach((k, v) -> {
            List<String> collums = v.stream()
                    .filter(column -> StringUtils.startsWith(column, "~"))
                    .collect(Collectors.toList());
            List<String> optionalTableColumns = ListUtils.subtract(v, collums);
            tableColumns.put(k, collums);
            optionalTableColumns.put(k, optionalTableColumns);

        });
        return new Diff(
                calculateSchemasDiff(expectedSchema, tableColumns),
                calculateSchemasDiff(expectedSchema, optionalTableColumns));
    }

    private class Diff {
        private final Map<String, List<String>> missingColumns;
        private final  Map<String, List<String>> optionalMissingColumns;

        public Diff(Map<String, List<String>> missingColumns, Map<String, List<String>> optionalMissingColumns) {

            this.missingColumns = missingColumns;
            this.optionalMissingColumns = optionalMissingColumns;
        }

        public Map<String, List<String>> getMissingColumns() {

            return missingColumns;
        }

        public Map<String, List<String>> getOptionalMissingColumns() {

            return optionalMissingColumns;
        }
    }

```

Reply:
another idea: can we cache JSON schemes Map<String, List<String>>? They are static, and they are not heavy.

Reply:
Do not see good reason for that but I've added it.